### PR TITLE
Initial DB migrations, also added to CI, and added some ITs that use the docker container

### PR DESCRIPTION
**Pull request by cfredericks**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,6 @@
+DB_USER=postgres
+DB_PASSWORD=password
+DB_NAME=postgres
+DB_HOST=localhost
+DB_PORT=15432
```
Comment (cfredericks) [inline .env:5]: this is for when we run a local DB with docker and set the port to foward to 15432

File: gcp/migrations/verify/appschema.sql [hunk 2]
```diff
@@ -0,0 +1,7 @@
+-- Verify golf-better:appschema on pg
+
+begin;
+
+select pg_catalog.has_schema_privilege('golfbetter', 'usage');
```
Comment (cfredericks) [inline gcp/migrations/verify/appschema.sql:5]: sqitch has these "verify" scripts in addition to the usual deploy+revert scripts. We can setup another workflow to verify our deployed DB as well to ensure the schema is in alignment (and also auto-deploy schema changes to dev), but that will be another PR

File: gcp/scripts/run_pg_refresh_pg_data_local_db.sh [hunk 3]
```diff
@@ -0,0 +1 @@
+DB_HOST=localhost DB_PORT=15432 DB_PASSWORD=password INSTANCE_CONNECTION_NAME= python3.11 ../cloud-functions/refresh-pga-data/main.py --write --data-types schedule leaderboard players
```
Comment (cfredericks) [inline gcp/scripts/run_pg_refresh_pg_data_local_db.sh:1]: some helper scripts for running things with a local DB (start_db.sh below starts the DB)

File: .github/workflows/refresh_pga_data_function.yml [hunk 4]
```diff
@@ -19,9 +35,35 @@ jobs:
...
+      working-directory: gcp/migrations
+      env:
+        PGUSER: postgres
+        PGPASSWORD: postgres
+        PGHOST: localhost
+        PGPORT: 5432
+        PGDATABASE: postgres
+      run: |
+        sqitch deploy db:pg://$PGUSER:$PGPASSWORD@$PGHOST:$PGPORT/$PGDATABASE
     - name: Test with pytest
```
Comment (cfredericks) [inline .github/workflows/refresh_pga_data_function.yml:58]: in a follow up PR i'll add some integration tests that actually use the DB, but this MR just sets everything up

File: gcp/cloud-functions/refresh-pga-data/test/integration/test_integration.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
+import main
+import sqlalchemy
+import time
+
+from datetime import datetime, date, timezone
+
+# TODO: Test other methods
```
Comment (cfredericks) [inline gcp/cloud-functions/refresh-pga-data/test/integration/test_integration.py:7]: todo in another PR

File: gcp/app-engine/test/integration/test_integration.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
+import main
+import json
+import pytest
+import sqlalchemy
+
+from datetime import datetime
+
+# TODO: Test other methods
```
Comment (cfredericks) [inline gcp/app-engine/test/integration/test_integration.py:8]: todo in another PR

File: .github/workflows/app_engine.yml [hunk 7]
```diff
@@ -19,10 +35,40 @@ jobs:
...
+        until pg_isready -h localhost -p 5432; do
+          echo "Waiting for Postgres to be ready..."
+          sleep 2
+        done
+    - name: Apply Sqitch migrations
+      working-directory: gcp/migrations
+      env:
+        PGUSER: postgres
+        PGPASSWORD: password
+        PGHOST: localhost
```
Comment (cfredericks) [inline .github/workflows/app_engine.yml:51]: from what i've read, using "postgres" here should work given the docker service name above, but the test was failing to resolve the host. Localhost seems to work though

**Review by kevlaube — APPROVED** [review]
lgtm
